internal/rivercommon/river_common.go

```diff
@@ -16,6 +16,8 @@ const (
 	QueueDefault       = "default"
 )
 
+type ContextKeyClient struct{}
```

Comment:
Thoughts on exposing a function here for setting the client on the context, rather than exposing the key itself? That would still make it impossible for that key to be utilized in any way other than by calling the function. And it feels more inline with the common pattern recommended in [the docs](https://pkg.go.dev/context#Context) (see `userKey`).

To make this work you would also need the getter function to be exposed in this package though  (`withClient`).

Reply:
Hm, unfortunately I think this turns out to be difficult because to get/set a client, you need a reference to the `Client` type. Having that in `rivercommon` would be a circular dependency, and even if we were to bring in a separate package for it, it'd still create an import cycle with the main package.

You saw that this is the internal `rivercommon` right? Should maybe it impossible to reuse `ContextKeyClient` for anything else.